src/box/replication.cc

```diff
@@ -206,6 +207,61 @@ replicaset_set_sync_quorum(void)
 	replication_sync_quorum_auto = replicaset.applier.connected;
 }
 
+/**
+ * Implementation of `replica_gc` that doesn't do any checks.
+ * Mustn't be called inside an active transaction.
+ */
+static int
+replica_gc_impl(struct replica *replica)
+{
+	assert(in_txn() == NULL);
+	if (gc_erase_consumer(&replica->uuid) != 0)
+		return -1;
+	if (replica->gc != NULL) {
+		gc_consumer_unregister(replica->gc);
+		replica->gc = NULL;
+	}
+	if (replica->gc_checkpoint_ref != NULL) {
+		gc_unref_checkpoint(replica->gc_checkpoint_ref);
+		replica->gc_checkpoint_ref = NULL;
+	}
+	/* Delete the replica if it is anonymous. */
+	if (replica->anon) {
+		replica_hash_remove(&replicaset.hash, replica);
+		replicaset.anon_count -= replica->anon;
+		assert(replicaset.anon_count >= 0);
+		replica_delete(replica);
+	}
+	return 0;
+}
+
+int
+replica_gc(const struct tt_uuid *uuid)
+{
+	if (in_txn() != NULL) {
+		diag_set(ClientError, ER_ACTIVE_TRANSACTION);
+		return -1;
+	}
+	if (tt_uuid_is_equal(uuid, &INSTANCE_UUID)) {
+		diag_set(ClientError, ER_REPLICA_GC, tt_uuid_str(uuid),
+			 "deactivation of self is not allowed");
```

Comment:
Sounds better, thanks! Also, now I start all the "details" messages with a capital letter since they are not the part of the main message.

By the way, updated `ER_REPLICA_GC` message: "Cannot cleanup replica's resources" -> "Cannot clean up replica's resources".